doc/reference/trd104-syscalls.md

```diff
@@ -554,16 +640,19 @@ platform and what drivers were compiled into the kernel.
 The register arguments for Command system calls are as follows. The registers
 r0-r3 correspond to r0-r3 on CortexM and a0-a3 on RISC-V.
 
-| Argument          | Register |
-|-------------------|----------|
-| Driver number     | r0       |
-| Command number    | r1       |
-| Argument 0        | r2       |
-| Argument 1        | r3       |
+| Argument          | Register | Type                         |
+|-------------------|----------|------------------------------|
+| Driver number     | r0       | `u32`                        |
+| Command number    | r1       | `u32`                        |
+| Argument 0        | r2       | `{u32, i32, u64_lo, i64_lo}` |
+| Argument 1        | r3       | `{u32, i32, u64_hi, i64_hi}` |
 
-Argument 0 and argument 1 are unsigned 32-bit integers. Command calls should
-never pass pointers: those are passed with Allow calls, as they can adjust
-memory protection to allow the kernel to access them.
+> When passing values derived from types whose representation is smaller than
+> the native machine register width (e.g. `u8`, `i16`), they must be explicitly
+> cast to a type whose storage matches the native machine register width in a
+> manner which compels the compiler to emit code which actually updates the
+> underlying storage. Signed values must use the native signed type, i.e., an
+> `i8` must not be cast to a `u32`.
```

Comment:
Why is this a quote?

Also, I don't think we can specify this based on what a compiler might do, because ultimately what needs to happen is the u32 containing something else (say a u8) must match what the capsule does to get the u8 from the u32.

```suggestion
When passing values with types whose representation is smaller than
32 bits (e.g. `u8` or `i16`), they must be explicitly
cast to a `u32` (if unsigned) or `i32` (if signed).
```